homeassistant/components/opnsense/config_flow.py

```diff
@@ -0,0 +1,130 @@
+"""Config flow for Opnsense integration."""
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_NAME,
+    CONF_SCAN_INTERVAL,
+    CONF_TIMEOUT,
+    CONF_URL,
+    CONF_VERIFY_SSL,
+)
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.selector import (
+    SelectSelector,
+    SelectSelectorConfig,
+    SelectSelectorMode,
+)
+
+from .const import (
+    CONF_API_SECRET,
+    CONF_TRACKER_INTERFACE,
+    DEFAULT_NAME,
+    DEFAULT_SCAN_INTERVAL,
+    DEFAULT_TIMEOUT,
+    DOMAIN,
+)
+
+
+class OPNSenseConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """OPNSense config flow."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle manual initiation of the config flow."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user",
+                data_schema=vol.Schema(
+                    {
+                        vol.Required(CONF_NAME, default=DEFAULT_NAME): str,
```

Comment:
We don't want to have names settable by the users in a config flow anymore except for when its a helper.

Reply:
Is there an example integration that follows this? So I know what needs changing other than removing this line.

Reply:
You have to make sure a user can't fill in a name, in that case you need to find something else for your config entry title. Preferably the actual device name, but some APIs don't return it, so then Host would work. As long as it's user identifiable.